### PR TITLE
API test for addUser

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
API test to add user in ownCloud server

## Related Issue
#31533 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -1225,6 +1225,7 @@ public function appShouldBeEnabled($app) {
 
 	/**
 	 * @Then /^user "([^"]*)" should be disabled$/
+	 * @When /^the user "([^"]*)" has been disabled$/
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php]: Actually this step is not an action, it is a test of the system state.
So we cannot write it as a ``When``.

File: tests/acceptance/features/apiProvisioning-v2/addUser.feature [hunk 2]
```diff
@@ -19,3 +19,11 @@ So that I can give people controlled individual access to resources on the ownCl
 		When the administrator sends a user creation request for user "brand-new-user" password "456newpwd" using the API
 		Then the OCS status code should be "400"
 		And the HTTP status code should be "400"
+
+	Scenario: Create an existing disabled user
+		Given user "brand-new-user" has been created
+		When user "admin" sends HTTP method "PUT" to API endpoint "/cloud/users/brand-new-user/disable"
+		And the user "brand-new-user" has been disabled
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/addUser.feature]: For these 2 steps together, we should write a step:
```
Given user "brand-new-user" has been disabled
```
and it should disable the user, and also check that the user is disabled.

**Review by phil-davis — COMMENTED** [review]
See comments

File: tests/acceptance/features/apiProvisioning-v2/addUser.feature (hunk 2 above)
Comment (paurakhsharma) [reply tests/acceptance/features/apiProvisioning-v2/addUser.feature]: Yes, I should do that

File: tests/acceptance/features/bootstrap/Provisioning.php (hunk 1 above)
Comment (paurakhsharma) [reply tests/acceptance/features/bootstrap/Provisioning.php]: Yeah, that does make sense. I will create a new function for it.

File: tests/acceptance/features/apiProvisioning-v2/addUser.feature [hunk 3]
```diff
@@ -19,3 +19,17 @@ So that I can give people controlled individual access to resources on the ownCl
...
+		And user "brand-new-user" has been disabled
+		When the administrator sends a user creation request for user "brand-new-user" password "456newpwd" using the API
+		Then the OCS status code should be "400"
+		And the HTTP status code should be "400"
+
+	Scenario: Admin creates a new user and adds him directly to a group
+		Given group "newgroup" has been created
+		When the administrator sends a user creation request for user "newuser" password "456firstpwd" group "newgroup" using the API
+		Then the OCS status code should be "100"
+		And the HTTP status code should be "200"
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/addUser.feature]: This is surprising. For v2 I expect the HTTP status code to be "400" or similar.

File: tests/acceptance/features/apiProvisioning-v1/addGroup.feature [hunk 4]
```diff
@@ -22,18 +22,10 @@ So that I can more easily manage access to resources by groups rather than indiv
 			| Finance (NP)        | Space and brackets                      |
 			| Admin&Finance       | Ampersand                               |
 			| admin:Pokhara@Nepal | Colon and @                             |
-			| maintenance#123     | Hash sign                               |
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/addGroup.feature]: The names with "#" and "%" will work after #31322 is merged. So you can rebase again after that, and remove these ``addGroup.feature`` changes.

**Review by phil-davis — COMMENTED** [review]
See comments

File: tests/acceptance/features/apiProvisioning-v1/addGroup.feature (hunk 4 above)
Comment (phil-davis) [reply tests/acceptance/features/apiProvisioning-v1/addGroup.feature]: #31322 is merged now. So you can rebase and take out the addGroup changes.

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on #31326

File: tests/acceptance/features/apiProvisioning-v2/addUser.feature (hunk 3 above)
Comment (patrickjahns) [reply tests/acceptance/features/apiProvisioning-v2/addUser.feature]: don't get me started - need to have better / proper api design to move forward in the future

**Pull request merged**